test/spec/Feature/Query/SpreadQueriesSpec.hs

```diff
@@ -112,3 +90,513 @@ spec =
+            {"factory":"Factory A","id":[1, 2],"name":["Process A1", "Process A2"],"factory_id":[1, 1],"category_id":[1, 2]},
+            {"factory":"Factory B","id":[3, 4],"name":["Process B1", "Process B2"],"factory_id":[2, 2],"category_id":[1, 1]},
+            {"factory":"Factory C","id":[5, 6, 7, 8],"name":["Process C1", "Process C2", "Process XX", "Process YY"],"factory_id":[3, 3, 3, 3],"category_id":[2, 2, 2, 2]},
+            {"factory":"Factory D","id":[],"name":[],"factory_id":[],"category_id":[]}
+          ]|]
+          { matchStatus  = 200
+          , matchHeaders = [matchContentTypeJson]
+          }
+      it "should select spread columns from a nested one-to-one relationship" $
+        get "/factories?select=factory:name,...processes(process:name,...process_costs(process_costs:cost))&order=name" `shouldRespondWith`
+          [json|[
+            {"factory":"Factory A","process":["Process A1", "Process A2"],"process_costs":[150.00, 200.00]},
+            {"factory":"Factory B","process":["Process B1", "Process B2"],"process_costs":[180.00, 70.00]},
+            {"factory":"Factory C","process":["Process C1", "Process C2", "Process YY", "Process XX"],"process_costs":[40.00, 70.00, 40.00, null]},
+            {"factory":"Factory D","process":[],"process_costs":[]}
+          ]|]
+          { matchStatus  = 200
+          , matchHeaders = [matchContentTypeJson]
+          }
+      it "should select spread columns from a nested many-to-one relationship" $
+        get "/factories?select=factory:name,...processes(process:name,...process_categories(categories:name))&order=name" `shouldRespondWith`
+          [json|[
+            {"factory":"Factory A","process":["Process A1", "Process A2"],"categories":["Batch", "Mass"]},
+            {"factory":"Factory B","process":["Process B2", "Process B1"],"categories":["Batch", "Batch"]},
+            {"factory":"Factory C","process":["Process YY", "Process XX", "Process C2", "Process C1"],"categories":["Mass", "Mass", "Mass", "Mass"]},
+            {"factory":"Factory D","process":[],"categories":[]}
+          ]|]
+          { matchStatus  = 200
+          , matchHeaders = [matchContentTypeJson]
+          }
+      it "should select spread columns from a nested one-to-many relationship" $
+        get "/factories?select=factory:name,...processes(process:name,...process_supervisor(supervisor_ids:supervisor_id))&order=name" `shouldRespondWith`
+          [json|[
+            {"factory":"Factory A","process":["Process A1", "Process A2"],"supervisor_ids":[[1], [2]]},
+            {"factory":"Factory B","process":["Process B1", "Process B2"],"supervisor_ids":[[3, 4], [1, 2]]},
+            {"factory":"Factory C","process":["Process C1", "Process C2", "Process XX", "Process YY"],"supervisor_ids":[[3], [3], [], []]},
+            {"factory":"Factory D","process":[],"supervisor_ids":[]}
+          ]|]
+          { matchStatus  = 200
+          , matchHeaders = [matchContentTypeJson]
+          }
```

Comment:
I was surprised by these nested arrays `"supervisor_ids":[[1], [2]]`.

Looking further, I guess it does make sense.. it's a pretty flexible syntax. We definitely need to document this double spread on one-to-many.